### DISCUSSION TITLE
Get number of elements in the queue ?

### DISCUSSION THREAD

**Discussion by dev31337** [cesanta/mongoose#2397] [Ask questions, get answers] [answered]
Hi,

Using mg_queue API, how is it possible to know how many messages are in the queue ? and how to how many free bytes are left ?

I think it could be done with mg_queue.tail/head but I fail to find how.

Note: There is an error in the documentation, in the example of [mg_queue_del()](https://mongoose.ws/documentation/#mg_queue_del) the call to mg_iobuf_del() shall be replaced with msg_queue_del().

Regards.

**Comment by scaprile** [comment, marked as answer]
Docs fixed, thanks!
We don't have the need to know that, so we did not care. It would require travelling the queue, thing you would actually do anyway when serving msgs.
https://github.com/cesanta/mongoose/blob/master/src/queue.c
_book() checks for room:
https://github.com/cesanta/mongoose/blob/57fd93eedaf9cd7a7bf6a5004c0e2d7059d39381/src/queue.c#L45-L56
